src/stcal/jump/jump.py

```diff
@@ -459,15 +507,110 @@ def find_ellipses(dqplane, bitmask, min_area):
     return ellipses
 
 
-def make_snowballs(jump_circles, sat_circles):
-    # Ths routine will create a list of snowballs (circles) that have the center of the saturation circle
-    # within the unclosing jump circle.
+def make_snowballs(gdq, integration, group, jump_ellipses, sat_ellipses, low_threshold, high_threshold,
+                   min_sat_radius, expansion, sat_flag, jump_flag):
+    # Ths routine will create a list of snowballs (ellipses) that have the center of the saturation circle
+    # within the enclosing jump rectangle.
     snowballs = []
-    for jump in jump_circles:
-        radius = jump[1]
-        for sat in sat_circles:
-            distance = np.sqrt((jump[0][0] - sat[0][0]) ** 2 + (jump[0][1] - sat[0][1]) ** 2)
-            if distance < radius:  # center of saturation is within the enclosing jump circle
-                if jump not in snowballs:
-                    snowballs.append(jump)
-    return snowballs
+    for jump in jump_ellipses:
+        sat_found = False
+        if near_edge(jump, low_threshold, high_threshold):
+            snowballs.append(jump)
+        else:
+            for sat in sat_ellipses:
+                # center of saturation is within the enclosing jump rectangle
+                if point_inside_ellipse(sat[0], jump):
+                    if jump not in snowballs:
+                        snowballs.append(jump)
+                        sat_found = True
+                        gdq[integration, :, :, :] = extend_saturation(gdq[integration, :, :, :],
+                                                                      group, [sat], sat_flag, jump_flag,
+                                                                      min_sat_radius, expansion)
+    return gdq, snowballs
+
+
+def point_inside_ellipse(point, ellipse):
+    delta_center = np.sqrt((point[0]-ellipse[0][0])**2 + (point[1]-ellipse[0][1])**2)
+    minor_axis = min(ellipse[1][0], ellipse[1][1])
+    if delta_center < minor_axis:
+        return True
+    else:
+        return False
+
+#  This routing tests whether the center of a jump is close to the edge of the detector.
+#  Jumps that are within the threshold will not requre a saturated core since this may be off the
+#  detector
+def near_edge(jump, low_threshold, high_threshold):
+    if jump[0][0] < low_threshold or jump[0][1] < low_threshold\
+        or jump[0][0] > high_threshold or jump[0][1] > high_threshold:
+        return True
+    else:
+        return False
+
+def find_faint_extended(data, gdq, read_noise_2d, nframes, snr_threshold=1.3, min_shower_area=40, inner=1,
```

Comment:
again here, since there are defaults in `spec` in jwst there shouldn't be any here so they dont get out of sync

Reply:
Again, it make the specification of the unit tests simplier.